task_manager.f95

```diff
@@ -0,0 +1,228 @@
+        n_tasks = 0
+        do i = 1, this%workers(w)%n_tasks
+          t = this%workers(w)%task_ids(i)
+          call print(i // ": " // t // ": " // this%tasks(t)%memory)
+          load = load + this%tasks(t)%memory
+        end do
+        call print("Summary: "//this%workers(w)%n_tasks//" tasks, "//load//" memory, "//this%workers(w)%n_padding//" padding")
+    end do
+  end subroutine task_manager_show_distribution
+
+  subroutine task_manager_export_distribution(this, only_id)
+    type(task_manager_type), intent(in) :: this
+    integer, optional :: only_id !> only this worker writes distribution
+
+    integer :: i, t, w
+    integer :: only_id_opt
+    type(Inoutput) :: file
+
+    only_id_opt = optional_default(UNDEFINED, only_id)
+    if (only_id_opt >= 0 .and. only_id_opt /= this%my_worker_id) return
+
+    call initialise(file, "tm.dist", OUTPUT)
+    call print("worker_id task_id_local task_id_global memory", file=file)
+    do w = 1, this%n_workers
+        do i = 1, this%workers(w)%n_tasks
+          t = this%workers(w)%task_ids(i)
+          call print(w // " " // i // " " // t // " " // this%tasks(t)%memory, file=file)
+        end do
+    end do
+    call finalise(file)
+  end subroutine task_manager_export_distribution
+
+  subroutine task_manager_init_tasks(this, n_tasks)
+    type(task_manager_type), intent(inout) :: this
+    integer, intent(in) :: n_tasks !> number of tasks to allocate
+
+    call print("Allocate %tasks for "//n_tasks//" tasks.")
+    allocate(this%tasks(n_tasks))
+  end subroutine task_manager_init_tasks
+
+  subroutine task_manager_init_workers(this, n_workers)
+    type(task_manager_type), intent(inout) :: this
+    integer, intent(in) :: n_workers !> number of workers to allocate and set
+
+    this%n_workers = n_workers
+    allocate(this%workers(this%n_workers))
+  end subroutine task_manager_init_workers
+
+  subroutine task_manager_deallocate(this)
+    type(task_manager_type), intent(inout) :: this
+
+    this%distributed = .false.
+    this%n_tasks = 0
+    this%n_workers = 0
+    if (allocated(this%tasks)) deallocate(this%tasks)
+    if (allocated(this%workers)) deallocate(this%workers)
+  end subroutine task_manager_deallocate
+
+  subroutine task_manager_print(this)
+    type(task_manager_type), intent(inout) :: this
+
+    call print_title('task_manager_type')
+    call print("%active: "//this%active)
+    call print("%my_worker_id: "//this%my_worker_id)
+    call print("%distributed: "//this%distributed)
+    call print("%unified_workload: "//this%unified_workload)
+    call print("%n_tasks: "//this%n_tasks)
+    call print("%n_workers: "//this%n_workers)
+    call print_title('MPI_obj')
+    call print(this%MPI_obj)
+    call print_title('ScaLAPACK_obj')
+    call print(this%ScaLAPACK_obj)
+    call print_title('task distrubtion')
+    call task_manager_show_distribution(this)
+    call print_title('')
+  end subroutine task_manager_print
+
+end module task_manager_module
```

Comment:
I am thinking about adding `if (.not. this%active) return` to every method but there would be no feedback that this happened as there is no return status. Any best practices/thoughts about that?

Reply:
print at `PRINT_VERBOSE` level?

Reply:
Good idea.

But the question concerns the whole module.

Reply:
Moved to QUIP.